### PR TITLE
Support actions change

### PR DESCRIPTION
**Pull request by oklex**
## Description of the change

Proposed change to simplify the way supported_actions are implemented.
This will reduce confusion with where the field should apply, and also simplifies the requests to work on the entire response data, and not just a subset.

P.S.
- I'm not sure if I updated schema.py properly

### CODE REVIEW

File: common/config_schema.go [hunk 1]
```diff
@@ -42,7 +42,10 @@ type RecordKey = struct {
 type RequiredFields = []SchemaKey
 
 type Validator = struct {
-	// for string chars (mutually exclusive)
+	// for events, string types (mutually exclusive)
+	WhiteList []string `json:"whitelist,omitempty" msgpack:"whitelist,omitempty"`
+	Blacklist []string `json:"blacklist,omitempty" msgpack:"blacklist,omitempty"`
+	// for string types (mutually exclusive)
```
Comment (oklex) [inline common/config_schema.go:48]: oops loops like I didn't branch off master - this is an older PR change

File: common/config_schema.go [hunk 2]
```diff
@@ -13,17 +13,6 @@ type SchemaObject struct {
 	ElementName        string `json:"element_name,omitempty" msgpack:"element_name,omitempty"`
 	ElementDescription string `json:"element_desc,omitempty" msgpack:"element_desc,omitempty"`
 
-	// legacy fields
-	// -------------------------------------------
```
Comment (oklex) [inline common/config_schema.go:17]: I've checked that we're not using these fields anymore on any extension and that we aren't referencing them anywhere on the front-end app

File: common/request_schema.go [hunk 3]
```diff
@@ -16,17 +16,32 @@ type StatusMessages struct {
...
+	// If this element is a type "Record" object, field "key" defines the keys that own "Fields"
+	Key RecordKey `json:"key,omitempty" msgpack:"key,omitempty"`
+	// what to call each element in the record/list - use for auto generated copy/labels
+	ElementName        string `json:"element_name,omitempty" msgpack:"element_name,omitempty"`
+	ElementDescription string `json:"element_desc,omitempty" msgpack:"element_desc,omitempty"`
+
+	// List of Requests that can be performed on the given
+	// response data. Will translate into buttons on elements that
+	// will issue a Request to Extension with the element's data
+	SupportedActions []ActionName `json:"supported_actions,omitempty" msgpack:"supported_actions,omitempty"`
```
Comment (oklex) [inline common/request_schema.go:31]: this is the where SupportedActions will live now!

**Review by josh-lc — COMMENTED** [review]
Git acknowledgement review

**Pull request closed without merging**